// marble-topbar/lib/Topbar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Topbar = undefined;

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _metalComponent = require('metal-component');

var _metalComponent2 = _interopRequireDefault(_metalComponent);

var _metalSoy = require('metal-soy');

var _metalSoy2 = _interopRequireDefault(_metalSoy);

var _metalToggler = require('metal-toggler');

var _metalToggler2 = _interopRequireDefault(_metalToggler);

var _metalState = require('metal-state');

var _TopbarSoy = require('./Topbar.soy.js');

var _TopbarSoy2 = _interopRequireDefault(_TopbarSoy);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * Topbar component.
 */
var Topbar = function (_Component) {
  _inherits(Topbar, _Component);

  function Topbar() {
    _classCallCheck(this, Topbar);

    return _possibleConstructorReturn(this, (Topbar.__proto__ || Object.getPrototypeOf(Topbar)).apply(this, arguments));
  }

  _createClass(Topbar, [{
    key: 'attached',
    value: function attached() {
      this.toggler = new _metalToggler2.default({
        content: '.topbar-list',
        header: '.topbar-toggle',
        expandedClasses: 'topbar-list-expanded'
      });
    }
  }, {
    key: 'disposed',
    value: function disposed() {
      var toggler = this.toggler;

      if (toggler) {
        toggler.dispose();
      }
    }
  }]);

  return Topbar;
}(_metalComponent2.default);

/**
 * State definition.
 * @static
 * @type {!Object}
 */


Topbar.STATE = {
  /**
   * Additional CSS classes to be added
   * @type {!String}
   * @default undefined
   */
  style: _metalState.Config.string(),

  /**
   * Defines how the logo should look like
   * @type {?Object|undefined}
   * @default undefined
   */
  logo: _metalState.Config.shapeOf({
    href: _metalState.Config.string(),
    icon: _metalState.Config.string(),
    image: _metalState.Config.string(),
    text: _metalState.Config.string()
  }),

  /**
   * The list of menu items
   * @type {?Array|undefined}
   * @default undefined
   */
  items: _metalState.Config.arrayOf(_metalState.Config.shapeOf({
    href: _metalState.Config.string(),
    label: _metalState.Config.string(),
    selected: _metalState.Config.bool(),
    variant: _metalState.Config.string()
  })).value([])
};

_metalSoy2.default.register(Topbar, _TopbarSoy2.default);

exports.Topbar = Topbar;
exports.default = Topbar;
