// electric-marble-components/lib/ElectricNavigation.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _metalToggler = require('metal-toggler');

var _metalToggler2 = _interopRequireDefault(_metalToggler);

var _metalSoy = require('metal-soy');

var _metalSoy2 = _interopRequireDefault(_metalSoy);

var _electricBaseComponents = require('electric-base-components');

var _electricBaseComponents2 = _interopRequireDefault(_electricBaseComponents);

var _metal = require('metal');

var _ElectricNavigationSoy = require('./ElectricNavigation.soy.js');

var _ElectricNavigationSoy2 = _interopRequireDefault(_ElectricNavigationSoy);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ElectricNavigation = function (_ElectricNavigationBa) {
  _inherits(ElectricNavigation, _ElectricNavigationBa);

  function ElectricNavigation() {
    _classCallCheck(this, ElectricNavigation);

    return _possibleConstructorReturn(this, (ElectricNavigation.__proto__ || Object.getPrototypeOf(ElectricNavigation)).apply(this, arguments));
  }

  _createClass(ElectricNavigation, [{
    key: 'attached',
    value: function attached() {
      if ((0, _metal.isServerSide)()) {
        return;
      }

      this.toggler = new _metalToggler2.default({
        content: '.' + this.listClasses,
        header: '.' + this.togglerClasses,
        expandedClasses: 'topbar-list-expanded'
      });
    }
  }, {
    key: 'disposed',
    value: function disposed() {
      var toggler = this.toggler;

      if (toggler) {
        toggler.dispose();
      }
    }
  }]);

  return ElectricNavigation;
}(_electricBaseComponents2.default);

ElectricNavigation.STATE = {
  listClasses: {},
  togglerClasses: {}
};

_metalSoy2.default.register(ElectricNavigation, _ElectricNavigationSoy2.default);

exports.default = ElectricNavigation;
